### PR TITLE
License cleanup + fix generator + remove garbage collection

### PR DESCRIPTION
**Pull request by mullermp**
I verified that this gem still works as intended after some tweaks. Has some misc cleanups, finish the expires TTL feature, and fix migration generation for rails 5+ (migrations are versioned).

I think the plan will be a slight rewrite and try to port this to aws-sdk-rails.

Inspired by #24 to get this gem updated and modern before a new release.

### CODE REVIEW

**Comment by mullermp** [comment]
inspired by #24 to get this updated. more to be done

File: lib/rails/generators/sessionstore/dynamodb/dynamodb_generator.rb [hunk 1]
```diff
@@ -46,9 +40,9 @@ def file_name
         name.underscore
       end
 
-      # @return [String] migration version using time stamp YYYYMMDDHHSS
-      def self.next_migration_number(dir = nil)
-        Time.now.utc.strftime("%Y%m%d%H%M%S")
+      # @return [String] activerecord migration version
+      def migration_version
```
Comment (alextwoods) [inline lib/rails/generators/sessionstore/dynamodb/dynamodb_generator.rb:44]: Is this being used? I dont see it in code - or is it used by Rail's somewhere?

File: spec/aws/session_store/dynamo_db/garbage_collection_spec.rb [hunk 2]
```diff
@@ -1,16 +1,3 @@
...
-#     http://aws.amazon.com/apache2.0/
-#
-# or in the "license" file accompanying this file. This file is
-# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
-# ANY KIND, either express or implied. See the License for the specific
-# language governing permissions and limitations under the License.
-
 require 'spec_helper'
 
 describe Aws::SessionStore::DynamoDB::GarbageCollection do
```
Comment (alextwoods) [inline spec/aws/session_store/dynamo_db/garbage_collection_spec.rb:3]: This can be removed entirely now right?

**Review by alextwoods — APPROVED** [review]
Nice!  Looks good.

File: lib/rails/generators/sessionstore/dynamodb/dynamodb_generator.rb (hunk 1 above)
Comment (mullermp) [reply lib/rails/generators/sessionstore/dynamodb/dynamodb_generator.rb:44]: It's used in the template `sessionstore_migration.rb`. The migrations are versioned after Rails 5 and inherit a different base class.

File: spec/aws/session_store/dynamo_db/garbage_collection_spec.rb (hunk 2 above)
Comment (mullermp) [reply spec/aws/session_store/dynamo_db/garbage_collection_spec.rb:3]: I forgot about tests - but yes I can delete those. I debated whether the garbage collection was useful even with the TTL feature. Thoughts?

Comment (alextwoods) [reply spec/aws/session_store/dynamo_db/garbage_collection_spec.rb:3]: Yeah - I think it makes sense to remove the garbage collection.  I think using TTL here is a better practice and we should encourage that.  Side note: we may want to add options for using ddb encryption.  I'd also potentially like to evaluate the pessimistic locking.... I'm not quite sure I understand the case for the fairly expensive pessimistic locking vs using a straightforward optimistic locking strategy.  But thats out of scope for this change.

Comment (mullermp) [reply spec/aws/session_store/dynamo_db/garbage_collection_spec.rb:3]: I think it makes sense to remove pessimistic locking as well.

**Pull request closed without merging**